BackendTaskPilot/Program.cs

```diff
@@ -0,0 +1,38 @@
+using System.Text.Json;
+using System.Text.Json.Serialization;
+
+var builder = WebApplication.CreateBuilder(args);
+var app = builder.Build();
+
+ExchangeRateService service = new ExchangeRateService();
+
+// REST API to get exchange rates based on input currencies
+app.MapPost("/", service.GetExchangeRates);
+
+app.Run();
+
+public struct CnbRatesResponse
```

Comment:
- You could create different layers to place the classes and separate the concerns. It could be folders or library classes. I'll simplify a lot, but an example could be:
1. _**Application**_: here goes the Controller and DTOs like **CnbRatesResponse**, and the new class that would represent the request body (**GetExchangeRatesRequest**)
2. **_Domain_**: here goes your entities and the **Rate** struct
3. **_Infrastructure_**: here goes everything related to external services, like the **ExchangeRateService** and **IExchangeRateService**